pkg/image/registry/image/strategy.go

```diff
@@ -129,6 +129,38 @@ func (imageStrategy) clearSignatureDetails(image *api.Image) {
 	}
 }
 
+// internalStrategy implements behaviour for signatures subset of image.
+type internalStrategy struct {
+	imageStrategy
+}
+
+// InternalStrategy is the default logic that applies when updating
+// image' signatures via the REST API.
+var InternalStrategy = internalStrategy{Strategy}
+
+// Canonicalize normalizes the object after validation.
+func (internalStrategy) Canonicalize(obj runtime.Object) {
+}
+
+// PrepareForUpdate currently allows to set just image signatures.
+func (s internalStrategy) PrepareForUpdate(obj, old runtime.Object) {
+	oldImage := old.(*api.Image)
+	image := obj.(*api.Image)
+
+	image.Labels = oldImage.Labels
+
+	image.DockerImageReference = oldImage.DockerImageReference
+	image.DockerImageMetadata = oldImage.DockerImageMetadata
+	image.DockerImageMetadataVersion = oldImage.DockerImageMetadataVersion
+	image.DockerImageManifest = oldImage.DockerImageManifest
+	image.DockerImageLayers = oldImage.DockerImageLayers
+	image.DockerImageSignatures = oldImage.DockerImageSignatures
+	image.DockerImageManifestMediaType = oldImage.DockerImageManifestMediaType
+	image.DockerImageConfig = oldImage.DockerImageConfig
+
+	s.clearSignatureDetails(image)
```

Comment:
> Can you add a comment for why you are doing this?

Yes, will do so.

> I think all the internalREST stuff goes away once we loopback to the API server instead of using a direct registry connection.

This should stay (at least in the other strategy) because we're not able to parse signatures yet to set signature metadata fields.